### PR TITLE
Discriminate window fit err msg from overdelegated error msg

### PR DESCRIPTION
**Pull request by felipemadero**
## Why this should be merged
Currently err msg for a delegation window not inside a validation window is given as overdelegation error

## How this works
Introduce new error kind and new checking function

### CODE REVIEW

**Review by abi87 — CHANGES_REQUESTED** [review]
nits

**Review by darioush — APPROVED** [review]

File: vms/platformvm/txs/executor/staker_tx_verification.go [hunk 1]
```diff
@@ -392,11 +393,15 @@ func verifyAddDelegatorTx(
 		return nil, err
 	}
 
-	canDelegate, err := canDelegate(chainState, primaryNetworkValidator, maximumWeight, newStaker)
+	fitsWindow := fitsValidationWindow(primaryNetworkValidator, newStaker)
+	if !fitsWindow {
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply vms/platformvm/txs/executor/staker_tx_verification.go:397]: Done

File: vms/platformvm/txs/executor/staker_tx_verification.go [hunk 2]
```diff
@@ -686,11 +691,15 @@ func verifyAddPermissionlessDelegatorTx(
 		return err
 	}
 
-	canDelegate, err := canDelegate(chainState, validator, maximumWeight, newStaker)
+	fitsWindow := fitsValidationWindow(validator, newStaker)
+	if !fitsWindow {
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply vms/platformvm/txs/executor/staker_tx_verification.go:695]: done

File: vms/platformvm/txs/executor/proposal_tx_executor.go [hunk 3]
```diff
@@ -700,26 +700,34 @@ func GetValidator(state state.Chain, subnetID ids.ID, nodeID ids.NodeID) (*state
...
+		return false
 	}
+	return true
+}
 
+// overdelegated returns true if [validator] will be overdelegated when adding [delegator].
+//
+// A [validator] would become overdelegated if:
+// - the maximum total weight on [validator] exceeds [weightLimit]
+func overdelegated(
```
Comment (StephenButtolph) [inline vms/platformvm/txs/executor/proposal_tx_executor.go:725]: golang uses camelCase.
```suggestion
func overDelegated(
```

**Review by StephenButtolph — COMMENTED** [review]
tests are also failing

File: vms/platformvm/txs/executor/staker_tx_verification.go [hunk 4]
```diff
@@ -686,11 +691,15 @@ func verifyAddPermissionlessDelegatorTx(
 		return err
 	}
 
-	canDelegate, err := canDelegate(chainState, validator, maximumWeight, newStaker)
+	fitsWindow := fitsValidationWindow(validator, newStaker)
```
Comment (joshua-kim) [inline vms/platformvm/txs/executor/staker_tx_verification.go:694]: Is it possible to add a unit test to verify this behavior now?

File: vms/platformvm/txs/executor/proposal_tx_executor.go (hunk 3 above)
Comment (felipemadero) [reply vms/platformvm/txs/executor/proposal_tx_executor.go:725]: changed

File: vms/platformvm/txs/executor/staker_tx_verification.go (hunk 4 above)
Comment (felipemadero) [reply vms/platformvm/txs/executor/staker_tx_verification.go:694]: it was considered in current unit tests, but they were also expecting the over delegation message

Comment (abi87) [reply vms/platformvm/txs/executor/staker_tx_verification.go]: if we reuse txs.BoundedBy function for delegators, we got the unit tests already

**Review by abi87 — APPROVED** [review]

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**